packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -642,12 +655,19 @@ export async function emitCode(context: vscode.ExtensionContext, uri: vscode.Uri
           }
         } else {
           resolve([...existingEmittersSelector.selectedItems]);
-          existingEmittersSelector.dispose();
+          existingEmittersSelector.hide();
```

Comment:
Was the issue that you immediately hide the notification here?

Reply:
I think the issue is because the onDidHide event was not handled in original logic which causes the quickpick can't be disposed properly when user click ESC which cause the whole progress status can't be closed properly.

Reply:
User's selection on the quick pick will trigger onDidAccpet event, but usually won't close the UI automatically. So need to call the hide() here to hide the quick pick explicitly which will then trigger the onDidHide event